pandas/core/arrays/masked.py

```diff
@@ -1440,10 +1440,8 @@ def _groupby_op(
 
         # libgroupby functions are responsible for NOT altering mask
         mask = self._mask
-        if op.kind != "aggregate":
-            result_mask = mask.copy()
-        else:
-            result_mask = np.zeros(ngroups, dtype=bool)
+
+        result_mask = np.zeros(ngroups, dtype=bool)
```

Comment:
This does not implement the change I suggested in the linked issue, and I believe is causing tests to fail.